Sources/BSON/ExtendedJSONEncoder.swift

```diff
@@ -0,0 +1,39 @@
+import Foundation
+/// Facilitates the encoding of `Encodable` values into ExtendedJSON.
+public class ExtendedJSONEncoder {
+    /// The enum representing the two possible encoding modes.
```

Comment:
I think it would be helpful to include some more info the docstrings here about the different between canonical and relaxed. you could take some inspiration from the definitions in the extJSON spec:
https://github.com/mongodb/specifications/blob/master/source/extended-json.rst#terms

Reply:
ok I added some more details about the different types and what they are for